connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -0,0 +1,277 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.planner
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.annotation.{Since, Unstable}
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.{expressions, plans}
+import org.apache.spark.sql.catalyst.analysis.{
+  UnresolvedAlias,
+  UnresolvedAttribute,
+  UnresolvedFunction,
+  UnresolvedRelation,
+  UnresolvedStar
+}
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, SubqueryAlias}
+import org.apache.spark.sql.catalyst.plans.logical
+import org.apache.spark.sql.types.{
+  BinaryType,
+  ByteType,
+  DateType,
+  DoubleType,
+  FloatType,
+  IntegerType,
+  ShortType,
+  TimestampType
+}
```

Comment:
I reopened the thread, @grundprinzip and @HyukjinKwon , because Apache Spark don't use this pattern yet.
- https://github.com/apache/spark/pull/37710#discussion_r978300187

```
% git grep '^import' | grep '.{$'
%
```

Reply:
I'm happy to change this in any way you want but can you please make your comment a bit more actionable?

- scalafmt produces an output that is validated in our style checks.
- I couldn't find anything in the Spark style guide on this topic that would ban an import like this.

Now my question is how do you want the output to look like and how do I produce the output automatically? It would be weird to request manual style adjustments that we don't verify in the build? (In particular since this is new code?)

Thanks in advance!

Reply:
Just saw https://github.com/apache/spark/pull/37710#discussion_r978300187.

I think this comment basically is to keep the code style consistent with the codebase that is (roughly) documented in https://spark.apache.org/contributing.html:

> If you’re not sure about the right style for something, try to follow the style of the existing codebase

So, yes, I think we should do some manual adjustments to be consistent in the current codebase in general, which I believe most of people do in their PRs.

Yes, ideally we should fix the Scala style guides to explicitly mention this.

Yes, `scalafmt` has an issue. There have been several discussions here and there, e.g., https://github.com/apache/spark/pull/35379#issuecomment-1033262392.

Reply:
Ok, I looked around and understand better now. Adjusted accordingly.